test/jdk/jdk/jfr/tool/TestMetadata.java

```diff
@@ -75,4 +87,65 @@ public static void main(String[] args) throws Throwable {
             lineNumber++;
         }
     }
-}
+
+    static void testEventTypeNum() throws Throwable {
+        OutputAnalyzer output = ExecuteHelper.jfr("metadata");
+        List<String> eventNames = new ArrayList<>();
+        List<String> lines = output.asLines();
+
+        for (String line : lines) {
+            if (line.startsWith("@Name(\"")) {
+                eventNames.add(line.substring(7, line.indexOf("\"", 7)));
+            }
+        }
+        List<EventType> eventTypes = FlightRecorder.getFlightRecorder().getEventTypes();
+        List<String> expectedNames = new ArrayList<>();
+        for (EventType eventType : eventTypes) {
+            expectedNames.add(eventType.getName());
+        }
+        Asserts.assertGTE(eventNames.size(), expectedNames.size());
+    }
+
+    static void testDeterministic() throws Throwable {
+        OutputAnalyzer output = ExecuteHelper.jfr("metadata", "--events", "CPULoad,GarbageCollection");
+        List<String> eventNames = new ArrayList<>();
+        List<String> lines = output.asLines();
+
+        for (String line : lines) {
+            if (line.startsWith("@Name(\"")) {
+                eventNames.add(line.substring(7, line.indexOf("\"", 7)));
+            }
+        }
+        Asserts.assertGTE(eventNames.size(), 2);
+    }
+
+    static void testWildcardAndAcronym() throws Throwable {
+        OutputAnalyzer output = ExecuteHelper.jfr("metadata", "--events", "Thread*");
```

Comment:
Instead of depending on name of existing JVM events (which may change, be removed or interfere with wildcard expansion done by the OS) it would be better to create two custom Java events with an unlikely name pattern, i.e. com.example.XXZQZYY and com.stuff.ZQZPP, and then filter on "ZQZ".  

To make sure they are registered, you can do:

    FlightRecorder.register(XXZQZYY.class); 
    FlightRecorder.register(ZQZPP.class); 

in static initializer so it is executed before calling ExecuteHelper.jfr("metadata", "--events", "ZQZ*");

Reply:
Make sense, I re-implement the test, using customized events.